python/igraphalg.py

```diff
@@ -0,0 +1,290 @@
+    target: mgp.Vertex,
+    edge_property: mgp.Nullable[str] = None,
+) -> mgp.Record(partition_vertex_ids=List[List[mgp.Vertex]], value=float):
+    graph, id_mapping, inverted_id_mapping = create_igraph_from_ctx(ctx, directed=True)
+    mincut = graph.mincut(
+        source=id_mapping[source.id],
+        target=id_mapping[target.id],
+        capacity=edge_property,
+    )
+
+    return mgp.Record(
+        partition_vertex_ids=[
+            convert_vertex_ids_to_mgp_vertices(partition, ctx, inverted_id_mapping)
+            for partition in mincut.partition
+        ],
+        value=mincut.value,
+    )
+
+
+@mgp.read_proc
+def topological_sort(
+    ctx: mgp.ProcCtx,
+) -> mgp.Record(nodes=mgp.Nullable[mgp.List[mgp.Vertex]]):
+    graph, _, inverted_id_mapping = create_igraph_from_ctx(ctx, directed=True)
+    sorted = graph.topological_sorting(mode="out")
+
+    return mgp.Record(
+        nodes=convert_vertex_ids_to_mgp_vertices(sorted, ctx, inverted_id_mapping)
+    )
+
+
+@mgp.read_proc
+def community_leiden(
+    ctx: mgp.ProcCtx,
+    edge_property: mgp.Nullable[str] = None,
+    resolution_parameter: float = 0.6,
+    number_of_iterations: int = -1,
+) -> mgp.Record(community_id=int, community_members=List[mgp.Vertex]):
+    graph, _, inverted_id_mapping = create_igraph_from_ctx(ctx, directed=False)
+
+    communities = graph.community_leiden(
+        resolution_parameter=resolution_parameter,
+        weights=edge_property,
+        n_iterations=number_of_iterations,
+    )
+
+    return [
+        mgp.Record(
+            community_id=i,
+            community_members=convert_vertex_ids_to_mgp_vertices(
+                members, ctx, inverted_id_mapping
+            ),
+        )
+        for i, members in enumerate(communities)
+    ]
+
+
+@mgp.read_proc
+def spanning_tree(
+    ctx: mgp.ProcCtx, edge_property: mgp.Nullable[str] = None, directed: bool = True
+) -> mgp.Record(tree=List[List[mgp.Vertex]]):
+    graph, id_mapping, inverted_id_mapping = create_igraph_from_ctx(
+        ctx, directed=directed
+    )
+    edge_weights = None
+    if edge_property:
+        edge_weights = graph.es[edge_property]
+
+    min_span_tree_graph = graph.spanning_tree(weights=edge_weights)
+    min_span_tree = get_min_span_tree_vertex_pairs(
+        ctx=ctx,
+        id_mapping=id_mapping,
+        inverted_id_mapping=inverted_id_mapping,
+        min_span_tree_graph=min_span_tree_graph,
+    )
+
+    return mgp.Record(tree=min_span_tree)
+
+
+@mgp.read_proc
+def shortest_path_length(
+    ctx: mgp.ProcCtx,
+    source: mgp.Vertex,
+    target: mgp.Vertex,
+    edge_property: mgp.Nullable[str] = None,
+    directed: bool = True,
+) -> mgp.Record(length=float):
+    graph, id_mapping, _ = create_igraph_from_ctx(ctx, directed=directed)
+
+    return mgp.Record(
+        length=graph.shortest_paths(
+            source=id_mapping[source.id],
+            target=id_mapping[target.id],
+            weights=edge_property,
+        )[0][0]
+    )
```

Comment:
Why are we taking only the first element here?

Reply:
Output of `shortest_paths` is 2D list that contains shortest path lengths between source nodes and destination nodes. In our case we specified source node and destination node so 2D list will have only one member but if didnt specify source and destination we would get 2D list that will contain shortest path length between each pair of nodes.

Reply:
It returns empty list.